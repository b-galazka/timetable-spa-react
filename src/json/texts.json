{
    "lastModified": "Ostatnia aktualizacja",
    "classroom": "sala",
    "schoolClass": "klasa",
    "title": {
        "base": "ZSCKP",
        "notFound": "błąd 404"
    },
    "number": "Nr",
    "hour": "Godz",
    "days": [
        "Poniedziałek",
        "Wtorek",
        "Środa",
        "Czwartek",
        "Piątek"
    ],
    "downloadApp": "Pobierz aplikację mobilną",
    "visitPage": "Przejdź do witryny",
    "goHomepage": "Przejdź na stronę główną"
}