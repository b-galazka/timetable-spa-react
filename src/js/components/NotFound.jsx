import React, { Component } from 'react';
import { withRouter, Link } from 'react-router-dom';
import propTypes from 'prop-types';

import errors from '../../json/errors';
import texts from '../../json/texts';

class NotFound extends Component {

    render() {

        return (
            <section className="app__no-content">
                <div>
                    <p>{errors.notFound}</p>

                    <Link
                        to="/"
                        className="button button--404"
                    >
                        {texts.goHomepage}
                    </Link>
                </div>
            </section>
        );
    }

    componentDidMount() {

        this.updateTitle();
    }

    updateTitle() {

        const { title } = texts;
        const { pathname } = this.props.location;

        if (pathname !== '/') {

            document.title = `${title.base} | ${title.notFound}`;
        }
    }
}

NotFound.propTypes = {
    // router
    location: propTypes.shape({ pathname: propTypes.string.isRequired }).isRequired
};

export default withRouter(NotFound);