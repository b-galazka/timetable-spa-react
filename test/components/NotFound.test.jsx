import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import NotFound from '../../src/js/components/NotFound';
import texts from '../../src/json/texts';

Enzyme.configure({ adapter: new Adapter() });

jest.mock(
    '../../src/json/texts',
    () => ({
        title: {
            base: 'base title',
            notFound: 'not found title'
        },
        goHomepage: 'go homepage'
    })
);

jest.mock('../../src/json/errors', () => ({ notFound: 'not found' }));

describe('NotFound component', () => {

    it('should render 404 page', () => {

        const tree = renderer.create(
            <MemoryRouter>
                <NotFound.WrappedComponent location={{ pathname: '/path' }} />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    describe('NotFound.prototype.componentDidMount', () => {

        let notFound;

        const title = 'old title';

        beforeEach(() => {

            notFound = shallow(
                <NotFound.WrappedComponent location={{ pathname: '/path' }} />
            ).instance();

            notFound.props = { ...notFound.props };

            document.title = title;
        });

        it('should update title if pathname is not equal to "/"', () => {

            const { title } = texts;

            notFound.componentDidMount();

            expect(document.title).toBe(`${title.base} | ${title.notFound}`);
        });

        it('should not update title if pathname is equal to "/"', () => {

            notFound.props.location.pathname = '/';

            notFound.componentDidMount();

            expect(document.title).toBe(title);
        });
    });
});