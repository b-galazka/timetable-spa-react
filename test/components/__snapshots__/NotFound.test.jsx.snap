// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render 404 page 1`] = `
<section
  className="app__no-content"
>
  <div>
    <p>
      not found
    </p>
    <a
      className="button button--404"
      href="/"
      onClick={[Function]}
    >
      go homepage
    </a>
  </div>
</section>
`;
